core/file/utime_spec.rb

```diff
@@ -16,30 +21,53 @@
 
   it "sets the access and modification time of each file" do
     File.utime(@atime, @mtime, @file1, @file2)
-    File.atime(@file1).to_i.should be_close(@atime.to_i, 2)
-    File.mtime(@file1).to_i.should be_close(@mtime.to_i, 2)
-    File.atime(@file2).to_i.should be_close(@atime.to_i, 2)
-    File.mtime(@file2).to_i.should be_close(@mtime.to_i, 2)
+    if @time_is_float
+      File.atime(@file1).should be_close(@atime, 0.0001)
+      File.mtime(@file1).should be_close(@mtime, 0.0001)
+      File.atime(@file2).should be_close(@atime, 0.0001)
+      File.mtime(@file2).should be_close(@mtime, 0.0001)
```

Comment:
What's the reason for this change? To test more precision on Windows?
I think that would be better as a separate extra example/test, and let this test be general and platform-independent (I think some systems only provide second resolution for `atime`, but I'm not sure anymore).

Reply:
> What's the reason for this change? To test more precision on Windows?

The change happened a while ago, and I don't recall if it was due to a change in Windows, newer (better) build tools, or what (it did require a commit or two in trunk, 2.5 at the time).  If it accidentally reverted back, it might cause issues.  Hence, the tightened tests.

It's patch I've been using in ruby-loco for a **long time**, I finally decided to move many of my test patches into 'core', as Windows is now fully tested on mswin & mingw.

> I think some systems only provide second resolution for atime, but I'm not sure anymore.

If it passed on Travis, I thought all was ok.  Thoughts?

Reply:
> If it passed on Travis, I thought all was ok. Thoughts?

It also needs to pass on all platforms of https://rubyci.org/
With the current code, the improved precision is only tested on Windows, so on the upside there are little risks.